frontend/app/custom-components/profile/edit/data-table-subscriptions.tsx

```diff
@@ -0,0 +1,218 @@
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  TableHead,
+  TableRow,
+  TableCell,
+} from "~/components/ui/table";
+import * as React from "react";
+
+import {
+  type ColumnDef,
+  type ColumnFiltersState,
+  flexRender,
+  getCoreRowModel,
+  getFilteredRowModel,
+  getSortedRowModel,
+  type SortingState,
+  useReactTable,
+  type VisibilityState,
+  type RowData,
+} from "@tanstack/react-table";
+import { Button } from "~/components/ui/button";
+import { ArrowUpDown, Plus } from "lucide-react";
+import { Input } from "~/components/ui/input";
+import { ScrollArea } from "~/components/ui/scroll-area";
+import { Switch } from "~/components/ui/switch";
+
+export interface MailingTableProps {
+  name: string;
+  dataArray: string[];
+}
+
+type DataRow = {
+  data: string;
+  active: boolean;
+};
+
+declare module "@tanstack/react-table" {
+  interface TableMeta<TData extends RowData> {
+    updateData: (
+      rowIndex: number,
+      columnId: keyof TData,
+      value: unknown,
+    ) => void;
+  }
+}
+
+export const getColumns = (name: string): ColumnDef<DataRow>[] => [
+  {
+    accessorKey: "data",
+    header: ({ column }) => {
+      return (
+        <Button
+          variant="ghost"
+          onClick={() => column.toggleSorting(column.getIsSorted() === "asc")}
+        >
+          {name}
+          <ArrowUpDown className="ml-2 h-4 w-4" />
+        </Button>
+      );
+    },
+    cell: ({ row }) => <div className="lowercase">{row.getValue("data")}</div>,
+  },
+  {
+    accessorKey: "active",
+    header: "Active",
+    cell: ({ row, table }) => (
+      <Switch
+        checked={row.getValue("active")}
+        onCheckedChange={(value) => {
+          table.options.meta?.updateData(row.index, "active", !!value);
+        }}
+        aria-label="Toggle active state"
+      />
+    ),
+    enableSorting: true,
+  },
+];
+
+export function DataTableSubsciptions({ name, dataArray }: MailingTableProps) {
+  const [sorting, setSorting] = React.useState<SortingState>([]);
+  const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>(
+    [],
+  );
+  const [columnVisibility, setColumnVisibility] =
+    React.useState<VisibilityState>({});
+  const [rowSelection, setRowSelection] = React.useState({});
+
+  const [internalData, setInternalData] = React.useState<DataRow[]>(() =>
+    dataArray.map((item) => ({
+      data: item,
+      active: true,
+    })),
+  );
+
+  React.useEffect(() => {
```

Comment:
```suggestion
 useEffect(() => {
```

you can directly import the hooks and use them like this. You dont have to change it everywhere, but lets do it like this from now on.